### PR TITLE
New SDLinearTractionTerm

### PR DESCRIPTION
**Pull request by vlukes**
- fix `geme_mulAVSB3py()` - values were calculated for one cell only
- new `SDLinearTractionTerm` - calculates shape sensitivity of  `LinearTractionTerm`
- new test for `SDLinearTractionTerm`

### CODE REVIEW

File: sfepy/terms/terms_surface.py [hunk 1]
```diff
@@ -18,9 +19,10 @@ class LinearTractionTerm(Term):
     :math:`\ul{f}` for a traction vector, and itself for a stress tensor.
 
     The material parameter can have one of the following shapes: 1 or (1, 1),
-    (D, 1), (S, 1). The symmetric tensor storage is used in the last case: in
-    3D S = 6 and the indices ordered as :math:`[11, 22, 33, 12, 13, 23]`, in 2D
-    S = 3 and the indices ordered as :math:`[11, 22, 12]`.
+    (D, 1), (S, 1), or (D, D) in the `eval` mode. The symmetric tensor storage
+    is used in the last case: in 3D S = 6 and the indices ordered as
```
Comment (rc) [inline sfepy/terms/terms_surface.py:23]: Make (S, 1) last so that it is the last case.

File: sfepy/terms/terms_surface.py [hunk 2]
```diff
@@ -95,6 +101,86 @@ def set_arg_types( self ):
...
+                  {'opt_material': None}]
+    integration = 'surface'
+
+    @staticmethod
+    def d_fun(out, traction, val, grad_mv, div_mv, sg):
+
+        tdim, tdim2 = (None, None) if traction is None else traction.shape[2:]
+        dim = val.shape[2]
+        sym = (dim + 1) * dim // 2
+        nel, nqp = div_mv.shape[:2]
```
Comment (rc) [inline sfepy/terms/terms_surface.py:134]: In `get_eval_shape()` there is `n_el`, `n_qp`...

**Comment by rc** [comment]
Thanks! It seems nobody used the tensor mode ;)

File: sfepy/terms/terms_surface.py (hunk 1 above)
Comment (vlukes) [reply sfepy/terms/terms_surface.py]: `(D, D)` is possible in  the `eval` mode only, so I modified the docstring in this way.

File: sfepy/terms/terms_surface.py (hunk 2 above)
Comment (vlukes) [reply sfepy/terms/terms_surface.py:134]: `SDLinearTractionTerm.d_fun()` is the static method so I would have to call `div_mv.get_data_shape()`, but it requires `integrals`, etc. I think the current solution is simple and reliable.

Comment (rc) [reply sfepy/terms/terms_surface.py:134]: Yes, just that you use different names for the same things (nel x n_el, ...)

File: sfepy/terms/terms_surface.py (hunk 1 above)
Comment (rc) [reply sfepy/terms/terms_surface.py]: OK, so what about
```
The symmetric tensor storage (S, 1) is as follows: in 3D S = 6 and the indices ordered as
```

Comment (vlukes) [reply sfepy/terms/terms_surface.py]: Yes, I see. I did not noticed "in the last case".

File: sfepy/terms/terms_surface.py (hunk 2 above)
Comment (vlukes) [reply sfepy/terms/terms_surface.py]: renamed: `nel`, `nqp` --> `n_el`, `n_qp`

**Comment by rc** [comment]
LGTM, thanks!

**Pull request merged**